frontend/src/types/BucketTreeNode.ts

```diff
@@ -0,0 +1,16 @@
+import type { Bucket } from '@/types';
+
+export type BucketTreeNode = {
+  key: string;
+  data: Bucket | BucketTreeNodeDummyData;
```

Comment:
It may generally be better to coerce this type to be one explicit type assignmenet instead of a union and defer the intersectioning interface to the child type instead - that way you should be able to theoretically avoid needing to do any casting in the code where we actually leverage this. See comment below.
```ts
data: BucketTreeNodeData
```